### PR TITLE
Deprecated the security context part of ExecutionParameters

### PR DESCRIPTION
**Pull request by pmahindrakar-oss**
Signed-off-by: Prafulla Mahindrakar <prafulla.mahindrakar@gmail.com>

# TL;DR

## Type
 - [ ] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [ ] Code completed
 - [ ] Smoke tested
 - [X] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
 _How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
_Remove the '*fixes*' keyword if there will be multiple PRs to fix the linked issue_

fixes https://github.com/flyteorg/flyte/issues/<number>

## Follow-up issue
_NA_
OR
_https://github.com/flyteorg/flyte/issues/<number>_

### CODE REVIEW

File: pkg/manager/impl/execution_manager_test.go [hunk 1]
```diff
@@ -3847,6 +3922,43 @@ func TestGetExecutionConfigOverrides(t *testing.T) {
...
+						K8SServiceAccount: launchPlanK8sServiceAccount,
+					},
+				},
+				MaxParallelism: launchPlanMaxParallelism,
+			},
+		}
+		execConfig, err := executionManager.getExecutionConfig(context.TODO(), request, launchPlan)
+		assert.NoError(t, err)
+		assert.Equal(t, launchPlanMaxParallelism, execConfig.MaxParallelism)
+		assert.Equal(t, launchPlanK8sServiceAccount, execConfig.SecurityContext.RunAs.K8SServiceAccount)
```
Comment (katrogan) [inline pkg/manager/impl/execution_manager_test.go:3958]: is this correct? if we set the deprecated field in the CreateExecutionRequest shouldn't that take precedence over whatever is in the launch plan?

File: pkg/manager/impl/execution_manager_test.go [hunk 2]
```diff
@@ -3922,7 +4063,29 @@ func TestGetExecutionConfigOverrides(t *testing.T) {
 		assert.Equal(t, rmLabels, execConfig.GetLabels().Values)
 		assert.Equal(t, rmAnnotations, execConfig.GetAnnotations().Values)
 	})
-	t.Run("matchable resource partial config", func(t *testing.T) {
+	t.Run("launchplan with no config except deprecated auth role but having matchable attribute", func(t *testing.T) {
```
Comment (katrogan) [inline pkg/manager/impl/execution_manager_test.go:4066]: also not sure about this case, if the launch plan has any overridable field set in the spec then shouldn't we get all the execution config attributes from the LP and not the matchable attribute?

File: pkg/manager/impl/execution_manager_test.go (hunk 1 above)
Comment (pmahindrakar-oss) [reply pkg/manager/impl/execution_manager_test.go:3958]: This gives security context the precedence if its sets on the launch plan.  

If its not set as in this test `request full config with deprecated auth role` then it uses the deprecated auth role.

Here the assumption is if user has migrated to launchplans with new way of passing security context then he should move to execution create requests also with the same or do we want to support this case and use the deprecated  fields from request

File: pkg/manager/impl/execution_manager_test.go (hunk 2 above)
Comment (pmahindrakar-oss) [reply pkg/manager/impl/execution_manager_test.go:4066]: Again the same arg here too i was thinking , 
If the user already is aware of using security context in one place then can we expect him to use in others too ?

File: pkg/manager/impl/execution_manager_test.go (hunk 1 above)
Comment (katrogan) [reply pkg/manager/impl/execution_manager_test.go:3958]: hm I see what you're saying wrt the user has migrated on their end from entirely using auth role to security ctx, i just don't know if we can safely make that assumption. perhaps i'm being too defensive. we could document this somewhere user-facing perhaps

**Review by katrogan — APPROVED** [review]

**Comment by pmahindrakar-oss** [comment]
Holding off on this since the existing fix worked . Will revisit this
cc : @katrogan

**Pull request closed without merging**